Lib/urllib/parse.py

```diff
@@ -469,6 +469,9 @@ def urlsplit(url, scheme='', allow_fragments=True):
         else:
             scheme, url = url[:i].lower(), url[i+1:]
 
+    _unsafe_chars_to_remove = ['\t', '\r', '\n']
```

Comment:
make this a module level constant rather than always assigning it and constructing the dict here on every call.

```
_UNSAFE_URL_BYTES_TO_REMOVE_TRANSTABLE = {
        _byte: None for _byte in b'\t\r\n'
}
```

Reply:
I concur that with @gpshead, but there are other alternatives: regular expressions and `str.replace()`.
```
url = re.sub('[\n\r\t]', '', url)
```
or
```
url = url.replace('\n', '').replace('\r', '').replace('\t', '')
```
The latter way seems the fastest in common case.
```
$ ./python -m timeit -s 'table = {b: None for b in b"\t\r\n"}; url = "http://python.org"' 'url.translate(table)'
200000 loops, best of 5: 873 nsec per loop
$ ./python -m timeit -s 'import re; p = re.compile("[\n\r\t]"); url = "http://python.org"' 'p.sub("", url)'
1000000 loops, best of 5: 366 nsec per loop
$ ./python -m timeit -s 'url = "http://python.org"' 'url.replace("\n", "").replace("\r", "").replace("\t", "")'
1000000 loops, best of 5: 181 nsec per loop
```
It can be even 2x more faster if write:
```
if '\n' in url: url = url.replace('\n', '')
if '\r' in url: url = url.replace('\r', '')
if '\t' in url: url = url.replace('\t', '')
```

Reply:
>  It can be even 2x more faster if write:

I will go with this. This hurts readability, but given this parser method is used at the lowest level, speed takes the precedence.

Reply:
heh, that isn't what i expected performance wise.  Thanks for checking Serhiy!